src/main/java/ayds/dodo/movieinfo/moredetails/model/repository/CardRepositoryImpl.kt

```diff
@@ -0,0 +1,54 @@
+package ayds.dodo.movieinfo.moredetails.model.repository
+
+import ayds.dodo.movieinfo.home.model.entities.OmdbMovie
+import ayds.dodo.movieinfo.moredetails.model.entities.EmptyCard
+import ayds.dodo.movieinfo.moredetails.model.entities.Card
+import ayds.dodo.movieinfo.moredetails.model.entities.Source
+import ayds.dodo2.tmdb.ExternalService
+import ayds.dodo.movieinfo.moredetails.model.repository.local.ExtraInfoLocalStorage
+import ayds.dodo2.tmdb.entities.*
+
+class CardRepositoryImpl(
+    private val localStorage: ExtraInfoLocalStorage,
+    private val externalService: ExternalService
+) : CardRepository {
+
+    override fun getCardInfo(movie: OmdbMovie): Card {
+        var movieInfo = localStorage.getMovieInfo(movie.title)
+
+        when{
+            movieInfo != null -> markMovieInfoAsLocal(movieInfo)
+            else ->{
+                movieInfo = try{
+                    tmdbResponseToTmdbMovieMapper(
+                        externalService.getMovieInfo(movie.title,movie.year))
+                } catch (e: Exception) {
+                    println("Error searching in external service " + e.message)
+                    EmptyCard
+                }
+                if(movieInfo != EmptyCard)
+                    localStorage.saveMovieInfo(movieInfo)
+            }
+        }
+        return movieInfo
+    }
+
+    private fun markMovieInfoAsLocal(movie: Card) {
+        movie.isLocallyStoraged = true
+    }
+
+    private fun tmdbResponseToTmdbMovieMapper(movieInfo: TmdbMovie): Card =
+        if (movieInfo == EmptyTmdbMovie) {
+            EmptyCard
+        }
+        else {
+            Card().apply {
+                this.title = movieInfo.title
+                this.plot = movieInfo.plot
+                this.imageUrl = movieInfo.imageUrl
+                this.source = movieInfo.source.toSource()
```

Comment:
Setear directamente la fuente 
```suggestion
                this.source = Source.TMDB
```